rasa/core/training/interactive.py

```diff
@@ -602,16 +630,38 @@ def _slot_history(tracker_dump: Dict[Text, Any]) -> List[Text]:
                 "value": a.get("action")}
                for a in predictions]
 
-    choices = ([{"name": "<create new action>", "value": OTHER_ACTION}] +
-               choices)
+    tracker = await retrieve_tracker(endpoint, sender_id)
+    evts = tracker.get("events", [])
+
+    session_actions_unique = []
+    session_actions_all = [a["name"] for a in _collect_actions(evts)]
+    [session_actions_unique.append(a) for a in session_actions_all if
+     a not in session_actions_unique]
+    old_actions = [action["value"] for action in choices]
+    new_actions = [{"name": action, "value": OTHER_ACTION + action} for action
+                   in session_actions_unique if action not in old_actions]
+
+    choices = ([{"name": "<create new action>", "value": NEW_ACTION}] +
+               new_actions + choices)
     question = questionary.select("What is the next action of the bot?",
                                   choices)
 
     action_name = await _ask_questions(question, sender_id, endpoint)
-    is_new_action = action_name == OTHER_ACTION
+    is_new_action = action_name == NEW_ACTION
 
     if is_new_action:
+        # create new action
         action_name = await _request_free_text_action(sender_id, endpoint)
+        if "utter_" in action_name:
+            utter_message = await _request_free_text_utterance(
+                sender_id, endpoint, action_name)
+            NEW_TEMPLATES[action_name] = {utter_message: "new message IL"}
+
+    elif action_name[:32] == OTHER_ACTION:
+        # action was newly created in the session, but not this turn
+        is_new_action = True
+        action_name = action_name[32:]
```

Comment:
not sure what this is going?

Reply:
We only want to ask the user to name the new action and possibly create an utter_template if the action is created in the same turn. But if the action was created in previous turns and already has a name and template we don't want to ask the user again. 
The action is still has to be classified as new since it is newly added in the session and else would run into an exception since it's not in the current domain file and cannot yet be executed.

Reply:
i guess i'm confused where the number 32 comes from?

Reply:
oh that's because the action name is `OTHER_ACTION + action`, where `OTHER_ACTION= uuid.uuid4().hex` which is a unique 32 digit identifier and `action` is the name of the action that the user choose. We need both information, the name and that it is a newly created action, so I simply stored this in the name, but the fact that it is a new action (OTHER_ACTION) is stored in `new_action = True` from this point on so it is dropped here.